api/Controllers/AvailableCarParkController.js

```diff
@@ -0,0 +1,31 @@
+const connectToDb = require('../Services/DbService')
+const carParkService = require('../Services/CarParkService')
+const jsonResponse = require('../Services/JsonResponseService')
+const DurationValidator = require('../Services/Validators/DurationValidator')
+
+let getAvailableCarParks = async (req, res) => {
+    try {
+        connectToDb(async (collection) => {
+            const duration = DurationValidator(req.query.duration)
+            let carParks = await carParkService.getAvailableCarParks(collection, duration)
+            if (carParks.length > 0) {
+                let jsonRes = jsonResponse.successful()
+                jsonRes.message = 'Success - car parks located'
+                jsonRes.data = carParks
+                res.json(jsonRes)
+            } else {
+                let jsonRes = jsonResponse.unsuccessful()
+                jsonRes.message = 'There are no car parks found'
+                jsonRes.status = 204
+                res.json(jsonRes)
+            }
```

Comment:
So this isn't a big deal, just something to think about - feel free to ignore this if you want!

You can avoid the else statement here as well, if you make it so both res.json() get returned, you won't need the else. 

This is a concept called 'early returning' that aims to avoid conditional logic and to improve readability.